File: ruby/Composite/block.rb
require_relative 'Group'

# Single block
class Block < Group
  attr_accessor :width
  attr_accessor :length
  attr_accessor :parent

  def initialize(width, length)
    @width = width
    @length = length
  end

  # assemble method should treat both single and group of blocks same.
  def assemble
    print "Adding a block of #{@width}X#{@length} \n"
  end
end

File: java/Composite/src/Block.java

// Single block
public class Block implements Group {
	
	int width;
	int length;
	Group parent;
	
    public Block(int width, int length) {
		super();
		this.width = width;
		this.length = length;
	}

 // assemble method should treat both single and group of blocks same.
    public void assemble() {
        System.out.println("Adding a block of "+ this.width +"X"+ this.length);
    }
}